light-client/src/predicates.rs

```diff
@@ -95,28 +95,37 @@ pub trait VerificationPredicates: Send {
         Ok(())
     }
 
-    /// Check that the given header is within the trusting period, adjusting for clock drift.
+    /// Check that the trusted header is within the trusting period, adjusting for clock drift.
     fn is_within_trust_period(
         &self,
-        header: &Header,
+        trusted_header: &Header,
         trusting_period: Duration,
+        now: Time,
+    ) -> Result<(), VerificationError> {
+        let expires_at = trusted_header.time + trusting_period;
+        ensure!(
+            expires_at > now,
```

Comment:
I wonder if need to account for clock drift here as well?

Reply:
As far as I remember the spec applies the clockdrift only to untrusted headers. If we change that here we should feed that back into the spec too:
https://github.com/informalsystems/tendermint-rs/blob/master/docs/spec/lightclient/verification/verification.md#lcv-func-valid1

Found this gem from the spec 🤣 
> If trusted.Header.Time > now - trustingPeriod the blabla

Reply:
A benefit to not adding the `clock_drift` is that we have some extra time to account for the time taken by the next verification steps. Also this could mean that if the `header` has already expired like a second ago but because we add the `clock_drift` there, it could satisfy the condition and consider it valid.

Reply:
I think we shouldn't need to worry about clock drift here since trusting_period is orders of magnitude greater than clock_drift but I'm not sure how best to surface that reasoning in the code (maybe just a comment). Also cc @josef-widder